clang/lib/Frontend/InitPreprocessor.cpp

```diff
@@ -475,6 +475,8 @@ static void InitializeStandardPredefinedMacros(const TargetInfo &TI,
   }
 
   if (LangOpts.SYCL) {
+    // DPCPP compiler identification macro
+    Builder.defineMacro("__DPCPP_COMPILER", "1");
```

Comment:
Should we instead name it `__DPCPP_COMPILER__` to be consistent with other identification macro identifiers like `__clang__` and `__STDC_VERSION__` etc?

Reply:
Also, should we pass a date here (like `202001L`) rather than a `1` so that users can distinguish between versions of the DPCPP compiler? e.g., someone may want to test which version of DPCPP so that they can enable (or disable) certain code paths through the preprocessor.

Reply:
> Also, should we pass a date here (like `202001L`) rather than a `1` so that users can distinguish between versions of the DPCPP compiler? e.g., someone may want to test which version of DPCPP so that they can enable (or disable) certain code paths through the preprocessor.

The plan here is to use __DPCPP_COMPILER for checking to see if it is a DPC++ compiler, and use other build-encoding macros in conjunction for versions of the compiler.

Reply:
It was pointed out that using compiler version macros creates quite messy and fragile code, and we want to try and discourage that. A big reason version macros are used is to check for features/extensions and there are better ways to do this - like __has_feature, __has_extension,  and __has_cpp_attribute, or even checking for the SYCL standard version.

Reply:
I'm not sure if SYCL features/extensions can currently be tested using the __has_feature/__has_extension macros though. This is probably a discussion we need to have.

Reply:
FYI: there was some discussion on that topic with SYCL community. Brief summary of that discussion - https://github.com/intel/llvm/issues/806#issuecomment-558109284 and [an alternative opinion](https://github.com/intel/llvm/issues/806#issuecomment-640112275).